### PR TITLE
fix: correctly align gitrest and historian r11s dependencies

### PR DESCRIPTION
**Pull request by znewton**
## Description

Gitrest and Historian both had multiple different R11s package dependency versions after #20243. This caused Lumberjack to be configured incorrectly.

## Reviewer Guidance

Any ideas how to proactively test for this?

### CODE REVIEW

File: server/gitrest/packages/gitrest-base/package.json [hunk 1]
```diff
@@ -52,15 +52,15 @@
 	},
 	"dependencies": {
 		"@fluidframework/common-utils": "^1.1.1",
-		"@fluidframework/gitresources": "^4.0.0",
-		"@fluidframework/protocol-base": "^4.0.0",
+		"@fluidframework/gitresources": "^5.0.0-250395",
+		"@fluidframework/protocol-base": "^5.0.0-250395",
```
Comment (tylerbutler) [inline server/gitrest/packages/gitrest-base/package.json:56]: You shopuldn't use range dependencies on prerelease versions. Use exact versions instead.

Comment (znewton) [reply server/gitrest/packages/gitrest-base/package.json:56]: Oh, my bad! Thanks for the call out

**Review by tylerbutler — CHANGES_REQUESTED** [review]
It looks like there's not a lot of CI protection preventing these mismatches. We could consider adding some policy or syncpack config to help prevent errors.

**Comment by tylerbutler** [comment]
Also, we should have a follow-up conversation about whether server changes that are needed in gitrest/historian should use a prerelease version or actually publish a release of server and consume it in client, gitrest, and historian. That would help prevent server changes from inadvertently "breaking" client or causing integration headaches that are unknown at the time the server change is made.

**Comment by alexvy86** [comment]
> Any ideas how to proactively test for this?

I'd start by asking if the split between `<package>` and `<package>-base` is strictly necessary. Withing the repo the only place where we seem to import the `-base` packages is the corresponding "real one", so I'd be inclined to merge them, and that would make the out-of-sync dependencies a non-issue, right?

If that's not possible, could the "real" packages detect this condition early (e.g. when calling `configureLogging()` right when they start) and immediately crash? That could be accompanied by a super simple smoke test of "can the container start successfully" when we are building them. Curtis and I discussed in the past that it would be a nice test to have, since it's happened before that a change broke the containers entirely (they got stuck in a CrashLoopBackoff cycle in AKS).

**Comment by alexvy86** [comment]
> I had also made a change a while back to use globals for Lumberjack instead of per-version instances, but that didn't prevent this. Not sure why, so I need to look into that, too.

It sounds exactly like [something I ran into with Axios and customizing the http agent](https://github.com/microsoft/FluidFramework/pull/18327/files#diff-94f75643f0794765ec040d58cd932d2e67a9413d8f14af68f7d357549c32e527R42). Basically, since there's two separate versions of the package (in this case Lumberjack) at play, the globals we're setting are only the globals for one of them.

> For the smoke test, I'm not sure how to technically check if a dependency depends on a different version of the same dependency within the runtime.

I wouldn't check for that specifically, I was thinking more of a runtime condition we could look for in the Lumberjack config itself. But now that I remember the details of the issue I linked above, I'm not sure that approach would be possible.

Another alternative we could explore is to have the base package expose a way to configure logging which would use _its_ version of Lumberjack instead of depending on the non-base package setting up globals. Something like passing the "global Lumberjack instance" to `configureLogging()` so we know it tweaks the "correct" one that will be used at runtime.

**Comment by alexvy86** [comment]
I'm also thinking that merging the base package and its "sibling" might be feasible, if we make the "start the actual service" something that needs to be called explicitly instead of happening a side effect of importing the package. FRS could just ignore that bit entirely, while our containers could run whatever command they needed to actually spin up the functional component.

**Comment by tylerbutler** [comment]
> @tylerbutler it would really be preferred if Gitrest and Historian were just part of Routerlicious monorepo so that we wouldn't be forced to release R11s packages constantly to get minor R11s package changes in Gitrest/Historian.

Do the gitrest and historian packages need to be published to public npm? If not, then we can change them to the @fluid-internal scope and publish them to the internal feeds only. That would eliminate my main objection to merging them into one release group.

**Review by alexvy86 — APPROVED** [review]
@tylerbutler not sure if the change you had requested was specifically the exact version for pre-release deps. If so, I think we shouldn't block this on additional reliability improvements, since the issue is completely preventing us from looking at any server errors in its logs (in our internal tests). At least we end up no worse than we are today in regard to this potentially happening.

**Review by tylerbutler — APPROVED** [review]
Let's follow up on whether we can change the scope of these packages to fluid-internal.

**Comment by tylerbutler** [comment]
@znewton @alexvy86 I opened https://github.com/microsoft/FluidFramework/pull/20524 to explore changing the scopes.

**Pull request merged**